plugins/aws-appmesh/plugin/commands.go

```diff
@@ -172,9 +175,11 @@ func (plugin *Plugin) setupEgressRules(
 		}
 	}
 
-	if config.EgressIgnoredPorts != "" {
-		err = iptable.Append("nat", egressChain, "-p", "tcp", "-m", "multiport", "--dports",
-			config.EgressIgnoredPorts, "-j", "RETURN")
+	if len(config.EgressIgnoredPorts) > 0 {
+		err = eachNSlice(config.EgressIgnoredPorts, 15, func(ports []string) error {
```

Comment:
Could we add comment on the reasoning behind splitting by 15? Often the context gets lost when we look at the code later.

Reply:
yeup i can add a comment

Reply:
continuing our thread https://github.com/aws/amazon-vpc-cni-plugins/pull/77#issuecomment-1266126223 here so its easier to follow. i dont think the e2e test you mentioned is maintained. this repository uses containernetworking library v0.8.1 but it looks like this test uses an earlier version resulting in compile errors. try running `go test ./... -tags=e2e_test` @ofiliz do you have any context? 

> https://github.com/containernetworking/cni/blob/v0.8.1/pkg/invoke/exec.go#L76

Reply:
I have run these tests recently. Could you try `make build e2e-test`

Reply:
hmm i see, weird. im still seeing the same errors. what version upgrades are you referring to? could you link me? i can look into fixing this if that is a priority but it is also my first interaction with this repo so any context would be greatly appreciated. on your recent runs of these tests were you able to verify that the test you referenced (https://github.com/aws/amazon-vpc-cni-plugins/blame/master/plugins/aws-appmesh/e2eTests/e2e_test.go) was run in that suite? bc from my testing it looks like it will be excluded. i had to change the go annotation since im on golang 1.8 but regardless i still see the same errors:

```
# github.com/aws/amazon-vpc-cni-plugins/plugins/aws-appmesh/e2eTests [github.com/aws/amazon-vpc-cni-plugins/plugins/aws-appmesh/e2eTests.test]
plugins/aws-appmesh/e2eTests/e2e_test.go:153:3: not enough arguments in call to invoke.ExecPluginWithoutResult
	have (string, []byte, *invoke.Args)
	want (context.Context, string, []byte, invoke.CNIArgs, invoke.Exec)
plugins/aws-appmesh/e2eTests/e2e_test.go:179:3: not enough arguments in call to invoke.ExecPluginWithResult
	have (string, []byte, *invoke.Args)
	want (context.Context, string, []byte, invoke.CNIArgs, invoke.Exec)
plugins/aws-appmesh/e2eTests/e2e_test.go:205:3: not enough arguments in call to invoke.ExecPluginWithoutResult
	have (string, []byte, *invoke.Args)
	want (context.Context, string, []byte, invoke.CNIArgs, invoke.Exec)
```

[appmesh-e2e-test-paulyeo.txt](https://github.com/aws/amazon-vpc-cni-plugins/files/9709801/appmesh-e2e-test-paulyeo.txt)

Reply:
I was referring to the last few commits on this repo where we upgraded multiple versions including CNI version. I only ran these tests before that. Could you post what errors are you seeing? Also is that happening for all plugins E2E test?

Reply:
ill DM you

Reply:
All tests in the package should build and pass as usual. I'll check if we broke something when we upgraded dependencies.